dev/make-distribution.sh

```diff
@@ -42,6 +42,8 @@ function exit_with_usage {
   echo "usage:"
   cl_options="[--name] [--tgz] [--mvn <mvn-command>]"
   echo "make-distribution.sh $cl_options <maven build options>"
+  echo "Decrease build times by adding the -T option: -T 1C will result in Maven using one thread per core."
+  echo "For example, dev/make-distribution.sh -T 1C --name hadoop-2.7.2 -Psparkr -Pyarn -Phadoop-2.7 -Phive -Phive-thriftserver -Dscala-2.11"
```

Comment:
I wanted to use --mvn -T 1C in the first place, but this gives

dev/make-distribution.sh: line 117: command: -T: invalid option
command: usage: command [-pVv] command [arg ...]
+ '[' '!' '' ']'
+ echo -e 'Could not locate Maven command: '\''-T'\''.'
Could not locate Maven command: '-T'.
+ echo -e 'Specify the Maven command with the --mvn flag'
Specify the Maven command with the --mvn flag
+ exit -1

when I try

```
dev/make-distribution.sh --name hadoop-2.7.2 --mvn -T 1C  -Pyarn -Phadoop-2.7 -Phive -Phive-thriftserver -Dscala-2.11
```

Given that MAVEN_OPTS are JVM args I think this is the best approach unless I'm using --mvn incorrectly

I'll remove the profiles as you're right this is really just to demonstrate multithreaded building, I see now that the community downloads only feature the Hadoop two digit version so, agreed, let's go with that here too for consistency